packages/weak_map/mod.test.ts

```diff
@@ -0,0 +1,234 @@
+      .toBe(
+        undefined,
+      );
+  });
+
+  it("should support proper iteration", () => {
+    const map = new IterableWeakMap<WeakKey, string>([
+      [{}, "value1"],
+      [{}, "value2"],
+    ]);
+
+    let size = 0;
+    for (const [key, value] of map) {
+      expect(key != null, "Key should not be null.").toBeTruthy();
+      expect(typeof value, "Value should be a string.").toBe("string");
+      size++;
+    }
+
+    expect(size, "Should iterate over 2 entries.").toBe(2);
+  });
+
+  it("should immediately reflect GC results (async)", async () => {
+    const map = new IterableWeakMap<WeakKey, string>();
+    let key = new Array(5e6).fill("*");
+    map.set(key, "value");
+
+    key = null!;
+    await sleep(0);
+    gc(); // Manually trigger garbage collection
+    await sleep(4);
+    expect(
+      map.size,
+      "The map should be empty after garbage collection.",
+    ).toBe(0);
+  });
+
+  it("should properly delete and clear items", () => {
+    const map = new IterableWeakMap<WeakKey, string>();
+    const key1 = {}, key2 = {};
+
+    map.set(key1, "value1").set(key2, "value2");
+    expect(map.size, "The map should initially have 2 entries.").toBe(2);
+
+    map.delete(key1);
+    expect(map.has(key1), "The map should no longer have key1.").toBe(false);
+    expect(map.size, "The map size should be 1 after deletion.").toBe(1);
+
+    map.clear();
+    expect(map.size, "The map should be empty after clearing.").toBe(0);
+  });
+
+  it("should throw on receipt of invalid keys", () => {
+    const map = new IterableWeakMap<WeakKey, string>();
+
+    expect(() => {
+      // deno-lint-ignore no-explicit-any
+      map.set("stringKey" as any, "value");
+    }).toThrow(TypeError);
+  });
+
+  it("should overwrite values for repeated keys", () => {
+    const map = new IterableWeakMap<WeakKey, string>();
+    const key = {};
+
+    map.set(key, "value1");
+    map.set(key, "value2"); // Reuse the same key
+
+    expect(
+      map.get(key),
+      "The map should update the value for a repeated key.",
+    ).toBe("value2");
+    expect(
+      map.size,
+      "The map size should remain 1 after updating a key's value.",
+    ).toBe(1);
+  });
+
+  it("should iterate properly after GC (async)", async () => {
+    const map = new IterableWeakMap<WeakKey, string>();
+    let key1 = new Array(5e6).fill("*"); // Large object to encourage GC
+    const key2 = {};
+
+    map.set(key1, "value1");
+    map.set(key2, "value2");
+
+    key1 = null!; // Make key1 eligible for GC
+    await sleep(0);
+    gc(); // Trigger garbage collection
+    await sleep(4);
+
+    expect(map.size, "The map should have 1 entry after GC.").toBe(1);
+
+    let count = 0;
+    for (const [key, _value] of map) {
+      assert(key != null, "Iterated key should not be null.");
+      count++;
+    }
+
+    assert(
+      count <= 2 && count >= 1,
+      "The map should iterate over at least 1 and at most 2 entries after GC.",
+    );
```

Comment:
As above, we should consistently use `expect` or `assert`, not both.

```suggestion
    expect(
      count <= 2 && count >= 1,
      "The map should iterate over at least 1 and at most 2 entries after GC.",
    ).toBe(true);
```